### PR TITLE
Decals will no longer exist in walls, lava, or space

### LINKED ISSUE #24041: Gib decals can land on space tiles
Generally happens during large explosions

### PR DESCRIPTION
**Pull request by Cyberboss**
Fixes #24041
:cl:
fix: Blood and other decals will no longer remain on turfs they shouldn't
/:cl:

### CODE REVIEW

File: code/game/atoms.dm [hunk 1]
```diff
@@ -457,6 +457,12 @@ var/list/blood_splatter_icons = list()
 //This proc is called on the location of an atom when the atom is Destroy()'d
 /atom/proc/handle_atom_del(atom/A)
 
+//called when the turf the atom resides on is ChangeTurfed
+/atom/proc/HandleTurfChange(turf/T)
+	for(var/I in src)
```
Comment (RemieRichards) [inline code/game/atoms.dm]: pls use lowercase a, not ```I```, typically its ```Typed Uppercase = untyped lowercase``` (```var/atom/A = a```) and i like that cos its pretty

File: code/game/atoms.dm [hunk 2]
```diff
@@ -457,6 +457,12 @@ var/list/blood_splatter_icons = list()
 //This proc is called on the location of an atom when the atom is Destroy()'d
 /atom/proc/handle_atom_del(atom/A)
 
+//called when the turf the atom resides on is ChangeTurfed
+/atom/proc/HandleTurfChange(turf/T)
+	for(var/I in src)
+		var/atom/A = I
+		A.handle_turf_change()
```
Comment (RemieRichards) [inline code/game/atoms.dm]: goofed your casing and missed out ```T``` arg

File: code/game/atoms.dm [hunk 3]
```diff
@@ -457,6 +457,12 @@ var/list/blood_splatter_icons = list()
 //This proc is called on the location of an atom when the atom is Destroy()'d
 /atom/proc/handle_atom_del(atom/A)
 
+//called when the turf the atom resides on is ChangeTurfed
+/atom/proc/HandleTurfChange(turf/T)
+	for(var/i in src)
```
Comment (RemieRichards) [inline code/game/atoms.dm]: but you didn't make it ```a``` ;_;

**Comment by coiax** [comment]
What about chasms?

**Comment by coiax** [comment]
Or water?

**Pull request merged**